src/ftxui/dom/text.cpp

```diff
@@ -27,80 +28,93 @@ class Text : public Node {
     int y = box_.y_min;
     if (y > box_.y_max)
       return;
-    for (wchar_t c : text_) {
-      const int width = wchar_width(c);
-      if (width >= 1) {
-        if (x > box_.x_max)
-          return;
-        screen.PixelAt(x, y).character = c;
-      } else {
-        screen.PixelAt(x - 1, y).character += c;
-      }
-      x += std::max(width, 0);
+    for (auto cell : Utf8ToGlyphs(text_)) {
+      if (x > box_.x_max)
+        return;
+      screen.PixelAt(x, y).character = cell;
+      ++x;
     }
   }
 
  private:
-  std::wstring text_;
+  std::string text_;
 };
 
 class VText : public Node {
  public:
-  VText(std::wstring text) : text_(text) {
-    for (auto& c : text_)
-      width_ = std::max(width_, wchar_width(c));
+  VText(std::string text) : text_(text) {
+    width_ = string_width(text_) ? 1 : 0;
   }
 
   void ComputeRequirement() override {
     requirement_.min_x = width_;
-    requirement_.min_y = text_.size();
+    requirement_.min_y = string_width(text_);
   }
 
   void Render(Screen& screen) override {
     int x = box_.x_min;
     int y = box_.y_min;
     if (x + width_ - 1 > box_.x_max)
       return;
-    for (wchar_t c : text_) {
+    for (const auto& it : Utf8ToGlyphs(text_)) {
       if (y > box_.y_max)
         return;
-      screen.at(x, y) = c;
+      screen.PixelAt(x, y).character = it;
       y += 1;
     }
   }
 
  private:
-  std::wstring text_;
+  std::string text_;
   int width_ = 1;
 };
 
-/// @brief Display a pieve of unicode text.
+/// @brief Display a piece of UTF8 encoded unicode text.
 /// @ingroup dom
 /// @see ftxui::to_wstring
 ///
 /// ### Example
 ///
 /// ```cpp
-/// Element document = text(L"Hello world!");
+/// Element document = text("Hello world!");
 /// ```
 ///
 /// ### Output
 ///
 /// ```bash
 /// Hello world!
 /// ```
-Element text(std::wstring text) {
+Element text(std::string text) {
   return std::make_shared<Text>(text);
 }
 
-/// @brief Display a pieve of unicode text vertically.
+/// @brief Display a piece of UTF8 encoded unicode text.
```

Comment:
This overload accepts a `std::wstring` which is not UTF-8 encoded. 

Same for `vtext(std::wstring)` below. 

Also, the doc string examples for these two overloads are missing L"" literals.

Reply:
Follow-up on the update - the vtext(std::string) overload doesn't mention an encoding.

Also, it might be cleaner to skip `UTF16` for `std::wstring` as *nix systems generally define 4-byte wchar_t.